packages/@react-stately/selection/src/useMultipleSelectionState.ts

```diff
@@ -59,15 +59,22 @@ export function useMultipleSelectionState(props: MultipleSelection): MultipleSel
     get childFocusStrategy() {
       return childFocusStrategyRef.current;
     },
-    setFocusedKey(k, childFocusStrategy = 'first') {
+    setFocusedKey(k, childFocusStrategy = 'first', callbackSet) {
       focusedKeyRef.current = k;
       childFocusStrategyRef.current = childFocusStrategy;
-      setFocusedKey(k);
+
+      if (callbackSet) {
+        for (let handler of callbackSet) {
```

Comment:
i think SelectionManager should be responsible for iterating over all the callbacks

Reply:
yeah... that would've been ideal, but the problem is as follows:
We set focused key in a bunch of places via `manager.setFocusedKey(key)` -> this calls `state.setFocusedKey(key)` using state obtained from a stately hook (for instance useGridState) -> useGridState overrides the key via https://github.com/adobe/react-spectrum/blob/main/packages/@react-stately/grid/src/useGridState.ts#L31-L46. This means if we have SelectionManager iterate over all the callbacks, we don't get the additional key overriding logic that useGridState does, resulting in inaccurate focus behavior. I'll think on this some more, lemme know if you have any ideas